src/catalog/src/durable/persist.rs

```diff
@@ -1221,6 +1222,44 @@ impl UnopenedPersistCatalogState {
             catalog
                 .increment_catalog_upgrade_shard_version(self.update_applier.organization_id)
                 .await;
+
+            // Before v0.120, the durable catalog opened a since handle using the
+            // controller's critical ID. Starting in v0.120, the durable catalog
+            // uses its own critical ID so that the storage controller can register
+            // a critical since handle using a controller critical ID. However, we
+            // need to expire our old critical since handle that uses the controller
+            // critical ID because it uses the wrong opaque type.
+            //
+            // TODO(benesch): remove this in v0.121.
```

Comment:
I haven't tried typing this, but if the following would work, I almost wonder if it's an easier hack to reason about.

- Somewhere before the storage controller gets this shard (here?) make sure the opaque value is at least 1
- Replace `assert_eq!(reader_state.opaque_codec, O::codec_name());` in State::CaDS with something that allows it to go from `"i64"` to `"PersistEpoch"`, which we can remove in the following release

Reply:
cc @bkirwi Would love any thoughts you have here

Reply:
Oh I thought about that, and I'm totally on board with that too! I was just worried that y'all would object to adding a hack in `mz-persist-client` like that because it applies to every shard, and it's really only the catalog shard that we want to allow this transition for.

Reply:
Yeah, it's for every shard, but it's pretty targeted (nothing else uses `i64` for Opaque) and it's easy to reason about things that happen in cmd impls (because linearized modifications to state, yadda yadda). I'd be a little stressed about the expire version because there are all sorts of larger distributed interactions there and if we hose some prod environment's CONTROLLER_CRITICAL_SINCE, then.. we're still gonna be pretty screwed :)